packages/mui-material/src/Select/SelectInput.js

```diff
@@ -324,6 +328,24 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     }
   };
 
+  const handleItemKeyDown = (child) => (event) => {
+    if (event.key === ' ') {
+      if (child.props.onKeyDown) {
+        child.props.onKeyDown(event);
+      }
```

Comment:
Why do we call the custom onKeyDown handler only on space, shouldn't this be outside of the condition?

Reply:
Oops sorry, i overlooked this. moved ` child.props.onKeyDown(event);` out of condition and modified test to reflect same here https://github.com/mui/material-ui/pull/43966/commits/e9f5d5b1d160cfe671b0bb44d15419081121d1a7